pkg/collector/python/init.go

```diff
@@ -278,48 +293,89 @@ func detectPythonLocation(pythonVersion string) {
 		PythonHome = pythonHome3
 	}
 
+	log.Infof("Using '%s' as Python home", PythonHome)
+
+	// For Windows, the binary should be in our path already and have a
+	// consistent name
 	if runtime.GOOS == "windows" {
-		pythonBinPath = filepath.Join(PythonHome, "python.exe")
-	} else {
-		// On Unix both python are installed on the same embedded
-		// directory. We don't want to use the default version (aka
-		// "python") but either "python2" or "python3" based on the
-		// configuration.
-		pythonBinPath = filepath.Join(PythonHome, "bin", "python"+pythonVersion)
+		// If we are in a development environment, PythonHome will not be set so we
+		// use the absolute path to the python.exe in our path.
+		if PythonHome == "" {
+			log.Warnf("Python home is empty. Inferring interpreter path from binary in path.")
+			interpreterAbsPath, err := resolveExecPath(PythonWinExeBasename)
+			if err != nil {
+				log.Warnf("Error trying to resolve interpreter path for executable: '%v'", err)
+			}
+
+			return interpreterAbsPath
+		}
+
+		return filepath.Join(PythonHome, PythonWinExeBasename)
 	}
+
+	// On *nix both Python versions are installed in the same embedded directory. We
+	// don't want to use the default version (aka "python") but rather "python2" or
+	// "python3" based on the configuration. Also on some Python3 platforms there
+	// are no "python" aliases either.
+	interpreterBasename := "python" + pythonVersion
+
+	// If we are in a development env or just the ldflags haven't been set, the PythonHome
+	// variable won't be set so what we do here is to just find out where our current
+	// default in-path "python2"/"python3" command is located and get its absolute path.
+	if PythonHome == "" {
+		interpreterAbsPath, err := resolveExecPath(interpreterBasename)
+		if err != nil {
+			log.Warnf("Error trying to resolve interpreter path for executable: '%v'", err)
+			return interpreterBasename
```

Comment:
A fatal error here would be inappropriate I think since almost everything would still work. The Python integrations would all still work it's just that the edge cases (like `multiprocessing` lib) would fail to run under `spawn` settings.

Reply:
If we're going to advertise `multiprocessing` as supported, then it does feel like it could be considered a fatal error to me.... At the very least we should then log something about what wouldn't work.